armi/reactor/parameters/parameterCollections.py

```diff
@@ -367,7 +367,7 @@ def __contains__(self, name):
         else:
             return name in self._hist
 
-    def __eq__(self, other):
+    def __eq__(self, other: "ParameterCollection"):
```

Comment:
Why is this a string?  `ParameterCollection` is a class defined in this file, wouldn't it be safe to just use the class name?

If I'm missing something here, just holler.

Reply:
Because it's not yet defined. We're in the definition of that class so python hasn't finished defining `ParameterCollection`.
```python
In [1]: class D:
   ...:     def __eq__(self, other: D):
   ...:         pass
   ...:
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 class D:
      2     def __eq__(self, other: D):
      3         pass

Cell In[1], line 2, in D()
      1 class D:
----> 2     def __eq__(self, other: D):
      3         pass

NameError: name 'D' is not defined
```